src/core_plugins/kibana/public/dashboard/reducers/panels.js

```diff
@@ -8,24 +8,53 @@ import {
   setPanels,
 } from '../actions';
 
-import { panel } from './panel';
-
 export const panels = handleActions({
-  [setPanels]: (state, { payload }) => _.cloneDeep(payload),
-  [updatePanels]: (state, { payload }) => {
-    const stateCopy = { ...state };
-    Object.values(payload).forEach(updatedPanel => {
-      stateCopy[updatedPanel.panelIndex] = _.defaultsDeep(updatedPanel, stateCopy[updatedPanel.panelIndex]);
-    });
-    return stateCopy;
-  },
-  [deletePanel]: (state, { payload }) => {
-    const stateCopy = { ...state };
-    delete stateCopy[payload];
-    return stateCopy;
-  },
-  [updatePanel]: (state, action) => ({
-    ...state,
-    [action.payload.panelIndex]: panel(state[action.payload.panelIndex], action),
-  }),
+  [setPanels]:
+    /**
+     *
+     * @param panels {Object.<string, PanelState>}
+     * @param payload {Object.<string, PanelState>}
+     * @return {Object.<string, PanelState>}
+     */
+    (panels, { payload }) => _.cloneDeep(payload),
+
+  [updatePanels]:
+    /**
+     *
+     * @param panels {Object.<string, PanelState>}
+     * @param payload {Object.<string, PanelState>}
+     * @return {Object.<string, PanelState>}
+     */
+    (panels, { payload }) => {
+      const panelsCopy = { ...panels };
+      Object.values(payload).forEach(updatedPanel => {
+        panelsCopy[updatedPanel.panelIndex] = _.defaultsDeep(updatedPanel, panelsCopy[updatedPanel.panelIndex]);
```

Comment:
This is super minor, but it might be nice to see this code shared with the `updatePanel` action below.

Reply:
lmk if the latest update is what you had in mind!

Reply:
🔥 💥